src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -10234,6 +10234,17 @@ static void Main()
 ");
         }
 
+        [Fact, WorkItem(64436, "https://github.com/dotnet/roslyn/issues/64436")]
+        public void SynthesizedDelegateTypes_NamedParameters()
```

Comment:
It'd be good to expand this test to cover a bit more of the cases described in "if any parameter or return value is not by value, or there are more than 16 parameters, or any of the parameter types or return are not valid type arguments (say, `(int* p) => { }`)".

Reply:
Right, thanks. I have consolidated and expanded the tests: d1a846d8bd48ce79f41edd91fe91a739c5553a0e